public/app/features/dashboard/services/TimeSrv.ts

```diff
@@ -148,8 +148,7 @@ export class TimeSrv {
   }
 
   private initTimeFromUrl() {
-    // If we are in a public dashboard ignore the time range in the url
-    if (config.isPublicDashboardView) {
+    if (this.timeModel?.timepicker.hidden) {
```

Comment:
When we were using `config.isPublicDashboardView` we were sure that this code would affect only publicDashboardViews, with this change, this could involve a regular dashboard as well (because that settings exists for dashboards). Are we sure we are not breaking anything here?

Reply:
Yes, you're right. If a dashboard disables the time range visibility, it's still possible to set a time range based on the url query params. I think it's the expected behavior, so I'm gonna add that public dashboard flag just to consider this case when it's a pubdash view